.github/workflows/deprecated.yml

```
@@ -0,0 +1,42 @@
+name: Deprecated ⚠️
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: "deprecated 할 버전 (예: 1.2.3)"
+        required: true
+        type: string
+      reason:
+        description: "deprecation 사유"
+        required: true
+        type: string
+
+jobs:
+  deprecated:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Validate version format
+        run: |
+          VERSION="${{ inputs.version }}"
+          if [[ ! "$VERSION" =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+            echo "❌ Invalid version format: $VERSION"
+            echo "Version must be in format: X.Y.Z (e.g., 1.2.3)"
+            exit 1
+          fi
+
+      - uses: actions/checkout@v6 # package.json 파일을 읽기 위해서 사용
+      - uses: actions/setup-node@v6
+        with:
+          registry-url: "https://registry.npmjs.org" # npmrc 자동 생성을 위해서 사용
+      - name: Deprecate npm package version
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+        run: |
+          if [ -z "$NODE_AUTH_TOKEN" ]; then
+            echo "❌ NODE_AUTH_TOKEN secret is not set"
+            exit 1
+          fi
+
+          PACKAGE_NAME=$(npm pkg get name | sed 's/"//g')
```

저는 이럴 때 `jq`가 더 간단해서 좋던데 효성님은 어떠실지 모르겠네요 ㅋ

```suggestion
          PACKAGE_NAME=$(jq -r .name package.json)
```